### PR TITLE
chore: Small refactor of `eval.rs`

### PR DESCRIPTION
**Pull request by Hofer-Julian**
# Description
- Extract long expression in `eval.rs` into variable
- Improve loop in eval.rs

# User-Facing Changes

None

# Tests + Formatting
✅

# After Submitting

### CODE REVIEW

File: crates/nu-engine/src/eval.rs [hunk 1]
```diff
@@ -1015,9 +1015,7 @@ pub fn eval_block(
     let num_pipelines = block.len();
 
     for (pipeline_idx, pipeline) in block.pipelines.iter().enumerate() {
-        let mut i = 0;
-
-        while i < pipeline.elements.len() {
+        for i in 0..pipeline.elements.len() {
```
Comment (sholderbach) [inline crates/nu-engine/src/eval.rs:1018]: Don't like the `i` for such a nest-fest.

As the whole thing cares about `pipeline.elements[i]` and checks `pipeline.elements[i+1]` I am wondering if here `elements.iter().peekable()` could simplify things.

Comment (Hofer-Julian) [reply crates/nu-engine/src/eval.rs:1018]: I already had that, but was unsure which one was nicer :D
Will go for the peekable then

**Comment by sholderbach** [comment]
Thanks for simplifying and cleaning this!

File: crates/nu-engine/src/eval.rs (hunk 1 above)
Comment (Hofer-Julian) [reply crates/nu-engine/src/eval.rs]: Done

Comment (sholderbach) [reply crates/nu-engine/src/eval.rs]: Lightning fast!

While it doesn't really save on line-real-estate, I think that this has less chances to mess things up.

Comment (Hofer-Julian) [reply crates/nu-engine/src/eval.rs]: > While it doesn't really save on line-real-estate, I think that this has less chances to mess things up.

Agreed.

**Comment by sholderbach** [comment]
Ready to ship this or is something else irking you here?

**Comment by Hofer-Julian** [comment]
> Ready to ship this or is something else irking you here?

Hah, ready from my side. If I find something else it will be a new PR

**Pull request merged**